docs/sql-getting-started.md

```diff
@@ -347,7 +347,8 @@ For example:
 </div>
 
 ## Scalar Functions
-(to be filled soon)
+
+Scalar functions are functions that return a single value per row, as opposed to aggregation functions, which return a value for a group of rows. Spark SQL supports a variety of built-in scalar functions such as [Array Functions](sql-ref-functions-builtin.html#array-functions), [Map Functions](sql-ref-functions-builtin.html#map-functions), [Date and Timestamp Functions](sql-ref-functions-builtin.html#date-and-timestamp-functions), etc. It also supports [User Defined Scalar Functions](sql-ref-functions-udf-scalar.html).
```

Comment:
Any reason not to describe aggregate/window functions? Also, I think its better not to describe each function group name here because we might forget to update this page every time we add a new group.

Reply:
I used Scalar function instead of built-in function (because the next paragraph is for aggregations). I don't want to talk about window function because I think window function is still sort of aggregation, even thought it returns result for each row.
I can't link it to builtin function because builtin function contains aggregation function, so I link to each group of the builtin scalar functions.

Reply:
I'm a bit confused; the next section describes the DataFrame aggregates in SQL documents?

Reply:
Seems  the next paragraph talks about builtin aggregation functions and user defined aggregation functions, so  I guess it's better for our paragraph to talk about scalar function than built-in function?

Reply:
Probably, we need to organize both paragraphs more. For example, how about this?

Adds two categories for built-in functions in `sql-ref-functions.html#built-in-functions`;
```
### Built-in Functions
Spark SQL has some ...
 
####  Scalar Functions 
Array Functions
Map Functions
Date and Timestamp Functions
JSON Functions

#### Aggregate-like Functions
Aggregate Functions
Window Functions
```

Then,
```
## Scalar Functions
A short description about scalar built-in functions and UDFs, then we add two links to their scalar sections.

## Aggregate Functions
A short description about aggregate built-in functions and UDFs, then we add two links to their aggregate sections.
```
Anyway, I think both categories had better to have same granularity description.

Reply:
OK... Seems Window functions still should be put in the Scalar function category. I will double check tomorrow.